### ISSUE TITLE
How can I set args like editable when using vue? 

### ISSUE THREAD

**Issue by programus** [TonyGermaneri/canvas-datagrid#260]
### Expected behavior and actual behavior.
I want the grid not editable, and I write this in vue template. 

```
    <canvas-datagrid
      class="data-grid"
      :data.prop="data"
      :schema.prop="schema"
      :editable.prop="false"
      ref="datagrid"
      :style.prop="{width: '100%', height: '100%'}"
    >
```

But I can still edit the value in cells. 

### Steps to reproduce the problem.
Here is the code snippet to reproduce the problem: https://codepen.io/programus/pen/jObPOBp 

### Specifications like the version of the project, operating system, or hardware.
macOS 10.14.6 Mojave
Chrome Version 80.0.3987.163 (Official Build) (64-bit)

**Comment by TonyGermaneri** [comment]
That really seems like it should work, sorry it does not.  Here's another way to disable edits:

On the grid template:

      @beginedit="endEdit"

Create an "endEdit" method and put this in it.

      this.$refs.datagrid.endEdit();

That should end the edit as it starts, effectively disabling edits.

Hope that works for you.

**Comment by programus** [comment]
@TonyGermaneri Thank you! It works! 

But actually, I also want to set some other options like allowSorting, autoResizeColumns, etc. 

I also updated the snippet which now have a read only grid but sorting is still allowed: https://codepen.io/programus/pen/jObPOBp

Is there any common way to solve this kind of problem once for all?

**Comment by TonyGermaneri** [comment]
add this to your template to prevent column sorting on click:

    @beforesortcolumn="$event.preventDefault()"

In spite of the name, `allowSorting` refers to the context menu option and not the entire grid, the documentation is wrong or this is a bug.


  mounted() {
    this.$refs.datagrid.attributes.allowSorting = false;
  },

Will allow you to get to all the members of the grid instance.

**Comment by TonyGermaneri** [comment]
I added a bug ticket to track the bad `allowSorting` property.  Sorry about that, thanks for finding it.

https://github.com/TonyGermaneri/canvas-datagrid/issues/261

**Comment by programus** [comment]
@TonyGermaneri Thank you! 

One more question, if I enable the sorting, once I clicked the header of a column it will be sorted ascending or descending, is there a way to reset the sorting to its original order? 

btw. I also found another maybe document bug here: https://tonygermaneri.github.io/canvas-datagrid/docs/index.html#tutorial--Format-data-using-rendertext-event
The code does not execute correctly, it would work if I used `formattext` instead of `rendertext` event.

**Comment by TonyGermaneri** [comment]
> is there a way to reset the sorting to its original order?

It's not tracked.  But you can hook into the event and track it yourself.  You can even add menu items to the context menu to reset it.

> btw. I also found another maybe document bug

That is a pretty shitty tutorial if it doesn't even work.  I imagine it hasn't worked for some time.  Thanks for finding another bug.  The academic part of my brain thanks you anyway.  The emotional part is starting to feel embarrassed.

**Comment by programus** [comment]
> > is there a way to reset the sorting to its original order?
> 
> It's not tracked. But you can hook into the event and track it yourself. You can even add menu items to the context menu to reset it.

I want to do this:
1. click the column header, it will sort ascending, and have a ▼ icon before header,
2. click the column header again, it will sort descending, and have a ▲ icon before header,
3. click the column header one more time, it will restore to the original order (I guess rowIndex have the information), and the icon disappears. 
4. click the column header again, same as No.1

I just read the document of `beforesortcolumn` and `sortcolumn`, (I think I found the correct events,) there are some information included in event object as input, but there seems no information about how to control the icon in the header. 
Could you give me a guide about how I can implement the 3-click sorting?

**Comment by programus** [comment]
@TonyGermaneri Is any advice for my problem above, or just it is not possible?

**Comment by TonyGermaneri** [comment]
Sorry, been busy.  No there's not really a way to do this right now without manually resetting the data from the original data source.  There is a ticket to fix this: #215 but it's pretty old.  You can get the behavior you desire by hooking into click events, and hooking into the arrow drawing events.  There are likely a few methods to achieve the behavior.

**Comment by superermiao** [comment]
why I can't use it in vue3?there is nothing to  show
